### PR TITLE
Hb05

### PR DESCRIPTION
**Pull request by tarkaiev**

### CODE REVIEW

File: src/main/java/movie/tickets/util/HashUtil.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        byte[] salt = new byte[16];
+        random.nextBytes(salt);
+        return salt;
+    }
+
+    public static String hashPassword(String password, byte[] salt) {
+
+        StringBuilder hashedPassword = new StringBuilder();
+
+        try {
```
Comment (dariiabutorina) [inline src/main/java/movie/tickets/util/HashUtil.java]:
```suggestion
    public static String hashPassword(String password, byte[] salt) {
        StringBuilder hashedPassword = new StringBuilder();
        try {
```

**Review by dariiabutorina — COMMENTED** [review]
GG, WP!

**Review by dariiabutorina — APPROVED** [review]

File: src/main/java/movie/tickets/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        byte[] salt = HashUtil.getSalt();
+        user.setPassword(HashUtil.hashPassword(user.getPassword(), salt));
+        user.setSalt(salt);
```
Comment (RomanDubovskyi) [inline src/main/java/movie/tickets/service/impl/UserServiceImpl.java]:
```suggestion
        user.setSalt(HashUtil.getSalt());
        user.setPassword(HashUtil.hashPassword(user.getPassword(), user.getSalt()));
```

File: src/main/java/movie/tickets/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDb = userService.findByEmail(email);
+        if (userFromDb.isEmpty()) {
```
Comment (RomanDubovskyi) [inline src/main/java/movie/tickets/security/AuthenticationServiceImpl.java]: let's shorten the code and use one if clause

File: src/main/java/movie/tickets/dao/impl/UserDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+            return user;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add new user", e);
```
Comment (RomanDubovskyi) [inline src/main/java/movie/tickets/dao/impl/UserDaoImpl.java]: let's make message more informative and add info about user

File: src/main/java/movie/tickets/security/AuthenticationServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            throw new AuthenticationException("No user with such login");
+        }
+        if (!HashUtil.hashPassword(password, userFromDb.get().getSalt())
+                .equals(userFromDb.get().getPassword())) {
+            throw new AuthenticationException("Incorrect password");
```
Comment (RomanDubovskyi) [inline src/main/java/movie/tickets/security/AuthenticationServiceImpl.java]: message should be "login or password is incorrect" it's more secured

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**